### DISCUSSION TITLE
Pytest cannot find module on docker container.

### DISCUSSION THREAD

**Discussion by cantos688** [pytest-dev/pytest#9060] [Q&A]
For various reasons at work, I have to be able to trigger pytest remotely via an api wrapper using in fast api. Tests are stored in multiple modules under a folder named test. I currently use this function to try to call a test module directly however no tests are collected.

```python
@app.get("/run_module/{test_module}", response_class=HTMLResponse)
def run_module(test_module:str,q: Optional[List[str]] = Query([])):
    try:
        html_file = "test_report.html"
        logger.info(os.getcwd())
        logger.info(f"test_module,{test_module}")
        logger.info( f"q : {str(q)}")
        module= Modules(name=test_module)
        output_str = " ".join(Pytest_Options(options=q).options)
        # run tests
        results = pytest.main(['--pyargs',f"~\app\test\{str(module.name).lower()}.py","--html="+html_file,"--self-contained-html", output_str])
        output = open(html_file, "r").read()

    except Exception as e:
        logger.info("Error during run_tests")
        logger.info(str(e))
        output = generate_html_error_response({"success": "false", "error": f"Run tests error: {str(e)}"})

    return output
```

Could anyone else advise a suggestion to how to invoke pytest main so that I can call a module directly?

**Comment by RonnyPfannschmidt** [comment]
in general its always suggested to run a subprocess

additionally for pyargs, a python module name needs to be passed, you passed a filename

**Comment by cantos688** [comment]
I started with the this as the first example that had that issue:
 pytest.main([f"{str(module.name).lower()}","--html="+html_file,"--self-contained-html", output_str])

**Reply by cantos688** [reply]
Also the sub-process isn't necessarily the best because in this case due to the only time the test cases are going to change is when the docker contain is being redeployed from a code merge from gitlab.

**Comment by nicoddemus** [comment]
The problem is that `--pyargs` expects an importable module string, for example `app.tests.test_foo`, but you are passing a file path `~\app\test\{str(module.name).lower()}.py`.

**Reply by cantos688** [reply]
Tried: f"app.test.{str(module.name).lower()}" and "app.test.test_foo"
Both didn't work.

**Reply by nicoddemus** [reply]
Sorry I should have been clearer:

If you just want to execute the test module directly, you don't need `--pyargs`:

```python
results = pytest.main([f"~\app\test\{str(module.name).lower()}.py", ...])
```

Just like you would pass the file directly in the command line:

```
pytest ~\app\test\test_foo.py
```

(`test_foo` above is just an example to what you receive as `test_module` in the endpoint)

**Reply by cantos688** [reply]
I need to check the repo, but I believe that was how i was calling it before and was also returning no results.
